pkg/filesystem/subvolume.go

```diff
@@ -52,6 +58,71 @@ func List(ctx context.Context, clientsets *k8sutil.Clientsets, operatorNamespace
 	listCephFSSubvolumes(ctx, clientsets, operatorNamespace, clusterNamespace, includeStaleOnly, subvolumeNames)
 }
 
+// checkForExternalStorage checks if the external mode is enabled.
+func checkForExternalStorage(ctx context.Context, clientsets *k8sutil.Clientsets, clusterNamespace string) bool {
+	enable := false
+	resource := "cephclusters"
+	cephclusters, err := clientsets.ListResourcesDynamically(ctx, "ceph.rook.io", "v1", resource, clusterNamespace)
+	if err != nil {
+		logging.Fatal(err)
+	}
+	if len(cephclusters) == 0 {
+		logging.Info("resource %s was not found on the cluster", resource)
+	}
+
+	for _, cephcluster := range cephclusters {
+		spec, _, err := unstructured.NestedMap(cephcluster.Object, "spec", "external")
+		if err != nil {
+			fmt.Printf("Error accessing 'externalStorage': %v\n", err)
+			os.Exit(1)
+		}
+		if spec["enable"] != nil {
+			enable = spec["enable"].(bool)
+			return enable
+		}
+	}
+	return enable
+}
+
+// getExternalClusterDetails exports the CEPH_ARGS with required mon-ip, id and key to connect to the
+// ceph cluster.
+func getExternalClusterDetails(ctx context.Context, clientsets *k8sutil.Clientsets, operatorNamespace, clusterNamespace string) (string, string, string) {
+	var admin_id, admin_key, m string
+
+	scList, err := clientsets.Kube.CoreV1().Secrets(clusterNamespace).List(ctx, v1.ListOptions{})
+	if err != nil {
+		logging.Fatal(fmt.Errorf("Error fetching secrets: %v\n", err))
+	}
+	for _, sc := range scList.Items {
+		if sc.ObjectMeta.Name == "rook-csi-cephfs-node" {
```

Comment:
why are we using cephfs node user? what if they are using someother secrets like restricted one? @parth-gr i hope its possible to use different secrets names for different filesystem/subvolumegroups

Reply:
Yes that's correct, it will be variable,
the prefix will be always `rook-csi-cephfs-node`, 
Not sure about the current operation but we can see the name from the storage class parameter.

Reply:
use provisioner secret and also check as per above comment we might not always create secret with exact names, you might need to use `strings.HasPrefix` or something similar